src/cache/cache.py

```diff
@@ -0,0 +1,21 @@
+import pandas as pd
+from src.data.dataset_loader import DatasetLoader
+
+class Cache(object):
+    def __init__(self, cache_dataset_name):
+        self.cache_dataset_name = cache_dataset_name
+        self.cache_dataset = DatasetLoader(cache_dataset_name)
+
+    def get(self, row: pd.DataFrame, field: str) -> str:
+        instruction =  row["instruction"]
+        match = self.cache_dataset.df.loc[self.cache_dataset.df['instruction'] == instruction, 'code']
```

Comment:
This should be passed in as the condition in the method instead of hard coding it here.

Reply:
This value is actually whether or not the row, code column is yes or no.